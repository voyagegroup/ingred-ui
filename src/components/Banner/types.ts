import * as React from "react";

/**
 * バナーのタイプを定義します
 * info: 情報提供用
 * warning: 警告用
 * error: エラー用
 */
export type BannerType = "info" | "warning" | "error";

/**
 * バナーのサイズを定義します
 * small: 小さいサイズ
 * medium: 中サイズ（デフォルト）
 */
export type BannerSize = "small" | "medium";

export type BannerProps = {
  /**
   * バナーのタイプ
   * @default "info"
   */
  type?: BannerType;
  /**
   * バナーのサイズ
   * @default "medium"
   */
  size?: BannerSize;
  /**
   * バナーに表示するメッセージ（childrenと併用する場合はこちらが優先されます）
   */
  message?: React.ReactNode;
  /**
   * バナーの内容（messageと併用する場合はmessageが優先されます）
   */
  children?: React.ReactNode;
  /**
   * 追加のCSSクラス
   */
  className?: string;
} & React.HTMLAttributes<HTMLDivElement>;
