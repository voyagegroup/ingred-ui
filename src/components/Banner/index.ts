export { default } from "./Banner";
export type { BannerProps, BannerType, BannerSize } from "./types";
