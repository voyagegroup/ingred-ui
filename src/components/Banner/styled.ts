import styled, { css } from "styled-components";
import { BannerType, BannerSize } from "./types";
import { fontSize } from "../Typography/Typography";

type ContainerProps = {
  type: BannerType;
  size: BannerSize;
};

type ContentWrapperProps = {
  size: BannerSize;
};

export const Container = styled.div<ContainerProps>`
  display: flex;
  width: 100%;
  /* サイズに応じたパディングの調整 */
  padding: ${({ theme, size }) =>
    size === "small" ? `${theme.spacing}px` : `${theme.spacing * 1.5}px`};
  border-radius: ${({ theme, size }) =>
    size === "small" ? `${theme.radius / 2}px` : `${theme.radius}px`};
  position: relative;


  /* タイプに基づいたスタイリング（背景色、テキスト色、ボーダー色） */
  ${({ theme, type }) => {
    switch (type) {
      case "info":
        return css`
          background-color: ${theme.palette.primary.ultraLight};
          color: ${theme.palette.primary.dark};
          border: 1px solid ${theme.palette.primary.light};
        `;
      case "warning":
        return css`
          background-color: ${theme.palette.warning.ultraLight};
          color: ${theme.palette.warning.deepDark};
          border: 1px solid ${theme.palette.warning.dark};
        `;
      case "error":
        return css`
          background-color: ${theme.palette.danger.ultraLight};
          color: ${theme.palette.danger.dark};
          border: 1px solid ${theme.palette.danger.light};
        `;
      default:
        return css`
          background-color: ${theme.palette.primary.ultraLight};
          color: ${theme.palette.primary.dark};
          border: 1px solid ${theme.palette.primary.light};
        `;
    }
  }}
`;

export const ContentWrapper = styled.div<ContentWrapperProps>`
  display: flex;
  align-items: center;
  /* サイズに応じたフォントサイズの調整 */
  font-size: ${({ size }) =>
    size === "small" ? `${fontSize.sm}px` : `${fontSize.md}px`};
  line-height: 1.4;
  color: inherit;
`;

export const CloseButton = styled.button`
  cursor: pointer;
  height: fit-content;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 4px;
  margin-left: ${({ theme }) => theme.spacing}px;
  background: none;
  border: none;
  border-radius: ${({ theme }) => theme.radius / 2}px;
  transition: background-color 0.2s;

  &:hover {
    background-color: color-mix(in srgb, currentColor 4%, transparent);
  }
`;
