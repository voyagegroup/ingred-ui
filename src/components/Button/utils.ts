import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
