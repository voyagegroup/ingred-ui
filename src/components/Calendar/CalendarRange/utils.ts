import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
