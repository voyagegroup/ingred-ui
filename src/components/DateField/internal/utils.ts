export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
