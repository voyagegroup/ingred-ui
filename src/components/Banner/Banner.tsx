import * as React from "react";
import * as Styled from "./styled";
import { useTheme } from "../../themes";
import Typography from "../Typography";
import Flex from "../Flex";
import Icon from "../Icon";
import { IconName } from "../Icon/Icon";
import { BannerProps } from "./types";

type AriaLive = "polite" | "assertive" | "off";

// タイプごとの設定をマッピングオブジェクトとして定義
const bannerTypeConfig = {
  info: {
    iconName: "information" as IconName,
    getColor: (theme: any) => theme.palette.primary.main,
    role: "status",
    ariaLive: "polite" as AriaLive,
  },
  warning: {
    iconName: "alert" as IconName,
    getColor: (theme: any) => theme.palette.warning.deepDark,
    role: "alert",
    ariaLive: "polite" as AriaLive,
  },
  error: {
    iconName: "alert" as IconName,
    getColor: (theme: any) => theme.palette.danger.main,
    role: "alert",
    ariaLive: "assertive" as AriaLive,
  },
};

const Banner = React.forwardRef<HTMLDivElement, BannerProps>(function Banner(
  {
    type = "info",
    size = "medium",
    message,
    children,
    className,
    closable = false,
    onClose,
    ...rest
  },
  ref,
) {
  const theme = useTheme();
  const config = bannerTypeConfig[type] || bannerTypeConfig.info;
  const iconName = config.iconName;
  const iconColor = config.getColor(theme);

  // messageとchildrenの両方が提供された場合、messageを優先
  const content = message || children;

  const handleClose = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();
    if (onClose) {
      onClose();
    }
  };

  return (
    <Styled.Container
      ref={ref}
      type={type}
      size={size}
      className={className}
      role={config.role}
      aria-live={config.ariaLive}
      {...rest}
    >
      <Flex
        display="flex"
        alignItems="center"
        justifyContent="space-between"
        style={{ width: "100%" }}
      >
        <Flex
          display="flex"
          alignItems="center"
          gap={size === "small" ? 1 : 1.5}
        >
          <Icon
            name={iconName}
            size={size === "small" ? "md-lg" : "lg"}
            color={iconColor}
          />
          {typeof content === "string" || typeof content === "number" ? (
            <Typography
              color="inherit"
              component="div"
              size={size === "small" ? "sm" : "md"}
            >
              {content}
            </Typography>
          ) : (
            <Styled.ContentWrapper size={size}>{content}</Styled.ContentWrapper>
          )}
        </Flex>

        {closable && (
          <Styled.CloseButton aria-label="閉じる" onClick={handleClose}>
            <Icon name="close" size="md" color={iconColor} />
          </Styled.CloseButton>
        )}
      </Flex>
    </Styled.Container>
  );
});

export default Banner;
