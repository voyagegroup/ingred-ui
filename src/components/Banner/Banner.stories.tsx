import React, { useState } from "react";
import { StoryObj } from "@storybook/react";
import Banner from "./Banner";
import { BannerProps } from "./types";
import Flex from "../Flex";
import Typography from "../Typography";
import Button from "../Button";

export default {
  title: "Components/Feedback/Banner",
  component: Banner,
  parameters: {
    docs: {
      description: {
        component:
          "バナーは、コンテキストに関連するメッセージを表示するためのコンポーネントです。" +
          "タイプに応じて適切なアクセシビリティ属性（role、aria-live）が自動的に設定されます。" +
          'Info: role="status", aria-live="polite"' +
          'Warning: role="alert", aria-live="polite"' +
          'Error: role="alert", aria-live="assertive"',
      },
    },
  },
  argTypes: {
    type: {
      description: "バナーのタイプ（info, warning, error）",
      control: { type: "select", options: ["info", "warning", "error"] },
      table: {
        type: { summary: "BannerType" },
        defaultValue: { summary: "info" },
      },
    },
    size: {
      description: "バナーのサイズ（small, medium）",
      control: { type: "select", options: ["small", "medium"] },
      table: {
        type: { summary: "BannerSize" },
        defaultValue: { summary: "medium" },
      },
    },
    message: {
      description: "バナーに表示するメッセージ（childrenより優先されます）",
      control: { type: "text" },
    },
    children: {
      description: "バナーの内容（messageが指定されている場合は無視されます）",
    },
    className: {
      description: "追加のCSSクラス",
    },
    closable: {
      description:
        "閉じるボタンを表示するかどうか（ConfirmModalと同様のスタイル）",
      control: "boolean",
      table: {
        type: { summary: "boolean" },
        defaultValue: { summary: "false" },
      },
    },
    onClose: {
      description: "閉じるボタンがクリックされたときのコールバック関数",
      action: "閉じるボタンがクリックされました",
    },
  },
};

const Template: StoryObj<BannerProps> = {
  render: (args) => <Banner {...args} />,
};

export const Info: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "info",
    message: "これは情報バナーです。ユーザーに情報を提供します。",
  },
};

export const Warning: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "warning",
    message: "これは警告バナーです。注意が必要な情報を提供します。",
  },
};

export const Error: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "error",
    message: "これはエラーバナーです。エラーが発生したことを示します。",
  },
};

export const Small: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "info",
    size: "small",
    message: "これは小さいサイズのバナーです。",
  },
};

export const Medium: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "info",
    size: "medium",
    message: "これは中サイズのバナーです。デフォルトサイズです。",
  },
};

export const Closable: StoryObj<BannerProps> = {
  ...Template,
  args: {
    type: "info",
    message:
      "これは閉じることができるバナーです。右側の×ボタンをクリックしてください。",
    closable: true,
  },
};

export const ClosableDemo: StoryObj = {
  render: () => {
    // eslint-disable-next-line react-hooks/rules-of-hooks
    const [visible, setVisible] = useState(true);

    const handleClose = () => {
      setVisible(false);
    };

    const handleReset = () => {
      setVisible(true);
    };

    return (
      <Flex display="flex" flexDirection="column" gap={2}>
        {visible ? (
          <Banner
            closable
            type="warning"
            message="これは閉じることができるバナーです。右側の×ボタンをクリックすると非表示になります。"
            onClose={handleClose}
          />
        ) : (
          <Button onClick={handleReset}>バナーを再表示</Button>
        )}
      </Flex>
    );
  },
};

export const WithImage: StoryObj<BannerProps> = {
  render: () => (
    <Banner closable type="info">
      <Flex display="flex" alignItems="center" gap={2}>
        <img
          src="https://placehold.jp/3d4070/ffffff/50x50.png"
          alt="サンプル画像"
          style={{ borderRadius: "4px" }}
        />
        <Typography>
          画像付きバナーの例です。任意のコンテンツを配置できます。
        </Typography>
      </Flex>
    </Banner>
  ),
};

export const RichContent: StoryObj<BannerProps> = {
  render: () => (
    <Banner closable type="warning">
      <Flex display="flex" flexDirection="column" gap={1}>
        <Typography size="lg" weight="bold">
          重要なお知らせ
        </Typography>
        <Typography>
          複数行のリッチなコンテンツを配置できます。レイアウトも自由に調整可能です。
        </Typography>
        <Flex display="flex" gap={1} alignItems="center">
          <span
            style={{
              textDecoration: "underline",
              cursor: "pointer",
              color: "inherit",
            }}
          >
            詳細を見る
          </span>
          <span>|</span>
          <span
            style={{
              textDecoration: "underline",
              cursor: "pointer",
              color: "inherit",
            }}
          >
            閉じる
          </span>
        </Flex>
      </Flex>
    </Banner>
  ),
};

export const AllVariants: StoryObj = {
  render: () => (
    <Flex display="flex" flexDirection="column" gap={2}>
      <Banner type="info" message="情報バナー" />
      <Banner type="warning" message="警告バナー" />
      <Banner type="error" message="エラーバナー" />
      <Banner type="info" size="small" message="小サイズのバナー" />
      <Banner type="warning" size="small" message="小サイズの警告バナー" />
      <Banner type="error" size="small" message="小サイズのエラーバナー" />
      <Banner closable type="info" message="閉じることができるバナー" />
      <Banner closable type="warning" message="閉じることができる警告バナー" />
      <Banner closable type="error" message="閉じることができるエラーバナー" />
    </Flex>
  ),
};
