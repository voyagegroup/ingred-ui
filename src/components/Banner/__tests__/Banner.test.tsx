import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import Banner from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("Banner component testing", () => {
  afterEach(cleanup);

  test("Banner with type info", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="info" message="Info message" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Banner with type warning", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="warning" message="Warning message" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Banner with type error", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="error" message="Error message" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Small size banner", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="info" size="small" message="Small size banner" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Medium size banner", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="info" size="medium" message="Medium size banner" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Banner with children", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="info">
        <div>Custom content inside banner</div>
      </Banner>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("Banner with complex children", () => {
    const { asFragment } = renderWithThemeProvider(
      <Banner type="warning">
        <div style={{ display: "flex", alignItems: "center", gap: "8px" }}>
          <img
            src="https://via.placeholder.com/24"
            alt="Icon"
            width="24"
            height="24"
          />
          <span>Content with image</span>
        </div>
      </Banner>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
