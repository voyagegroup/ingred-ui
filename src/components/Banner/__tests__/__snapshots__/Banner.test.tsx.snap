// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component testing Banner with children 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi bXaQjd"
    role="status"
    type="info"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fill="none"
            />
            <path
              d="M11 7V5H9V7H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              d="M11 14V9H9V14H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              clip-rule="evenodd"
              d="M10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0ZM2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10Z"
              fill="#0B82F4"
              fill-rule="evenodd"
              transform="translate(2 2)"
            />
          </svg>
        </span>
        <div
          class="sc-dkzDqf dyrjBY"
        >
          <div>
            Custom content inside banner
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Banner with complex children 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi geveWj"
    role="alert"
    type="warning"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,0H24V24H0Z"
              fill="none"
            />
            <path
              d="M12.866,3l9.526,16.5a1,1,0,0,1-.866,1.5H2.474a1,1,0,0,1-.866-1.5L11.134,3a1,1,0,0,1,1.732,0ZM4.206,19H19.794L12,5.5,4.206,19ZM11,16h2v2H11Zm0-7h2v5H11Z"
              fill="#785F03"
            />
          </svg>
        </span>
        <div
          class="sc-dkzDqf dyrjBY"
        >
          <div
            style="display: flex; align-items: center; gap: 8px;"
          >
            <img
              alt="Icon"
              height="24"
              src="https://via.placeholder.com/24"
              width="24"
            />
            <span>
              Content with image
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Banner with type error 1`] = `
<DocumentFragment>
  <div
    aria-live="assertive"
    class="sc-gsnTZi fIQyjq"
    role="alert"
    type="error"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,0H24V24H0Z"
              fill="none"
            />
            <path
              d="M12.866,3l9.526,16.5a1,1,0,0,1-.866,1.5H2.474a1,1,0,0,1-.866-1.5L11.134,3a1,1,0,0,1,1.732,0ZM4.206,19H19.794L12,5.5,4.206,19ZM11,16h2v2H11Zm0-7h2v5H11Z"
              fill="#EB0A4E"
            />
          </svg>
        </span>
        <div
          class="sc-bczRLJ bdqCNN"
          color="inherit"
          font-size="14px"
        >
          Error message
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Banner with type info 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi bXaQjd"
    role="status"
    type="info"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fill="none"
            />
            <path
              d="M11 7V5H9V7H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              d="M11 14V9H9V14H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              clip-rule="evenodd"
              d="M10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0ZM2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10Z"
              fill="#0B82F4"
              fill-rule="evenodd"
              transform="translate(2 2)"
            />
          </svg>
        </span>
        <div
          class="sc-bczRLJ bdqCNN"
          color="inherit"
          font-size="14px"
        >
          Info message
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Banner with type warning 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi geveWj"
    role="alert"
    type="warning"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0,0H24V24H0Z"
              fill="none"
            />
            <path
              d="M12.866,3l9.526,16.5a1,1,0,0,1-.866,1.5H2.474a1,1,0,0,1-.866-1.5L11.134,3a1,1,0,0,1,1.732,0ZM4.206,19H19.794L12,5.5,4.206,19ZM11,16h2v2H11Zm0-7h2v5H11Z"
              fill="#785F03"
            />
          </svg>
        </span>
        <div
          class="sc-bczRLJ bdqCNN"
          color="inherit"
          font-size="14px"
        >
          Warning message
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Medium size banner 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi bXaQjd"
    role="status"
    type="info"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ mdqMp"
        display="flex"
      >
        <span
          class="sc-jSMfEi llQWsy"
          size="24"
        >
          <svg
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fill="none"
            />
            <path
              d="M11 7V5H9V7H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              d="M11 14V9H9V14H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              clip-rule="evenodd"
              d="M10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0ZM2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10Z"
              fill="#0B82F4"
              fill-rule="evenodd"
              transform="translate(2 2)"
            />
          </svg>
        </span>
        <div
          class="sc-bczRLJ bdqCNN"
          color="inherit"
          font-size="14px"
        >
          Medium size banner
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Banner component testing Small size banner 1`] = `
<DocumentFragment>
  <div
    aria-live="polite"
    class="sc-gsnTZi cploy"
    role="status"
    type="info"
  >
    <div
      class="sc-eCYdqJ ljAMHk"
      display="flex"
      style="width: 100%;"
    >
      <div
        class="sc-eCYdqJ jEaNgF"
        display="flex"
      >
        <span
          class="sc-jSMfEi cLFvnm"
          size="20"
        >
          <svg
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              fill="none"
            />
            <path
              d="M11 7V5H9V7H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              d="M11 14V9H9V14H11Z"
              fill="#0B82F4"
              transform="translate(2 2)"
            />
            <path
              clip-rule="evenodd"
              d="M10 0C4.47715 0 0 4.47715 0 10C0 15.5228 4.47715 20 10 20C15.5228 20 20 15.5228 20 10C20 4.47715 15.5228 0 10 0ZM2 10C2 5.58172 5.58172 2 10 2C14.4183 2 18 5.58172 18 10C18 14.4183 14.4183 18 10 18C5.58172 18 2 14.4183 2 10Z"
              fill="#0B82F4"
              fill-rule="evenodd"
              transform="translate(2 2)"
            />
          </svg>
        </span>
        <div
          class="sc-bczRLJ hEOzTa"
          color="inherit"
          font-size="13px"
        >
          Small size banner
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
